src/HttpUserAgentParser/HttpUserAgentParser.cs

```diff
@@ -143,4 +175,62 @@ public static bool TryGetMobileDevice(string userAgent, [NotNullWhen(true)] out
         device = GetMobileDevice(userAgent);
         return device is not null;
     }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool ContainsIgnoreCase(ReadOnlySpan<char> haystack, ReadOnlySpan<char> needle)
+        => TryIndexOf(haystack, needle, out _);
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool TryIndexOf(ReadOnlySpan<char> haystack, ReadOnlySpan<char> needle, out int index)
+    {
+        index = haystack.IndexOf(needle, StringComparison.OrdinalIgnoreCase);
+        return index >= 0;
+    }
+
+    /// <summary>
+    /// Extracts a dotted numeric version.
+    /// Accepts digits and dots; skips common separators ('/', ' ', ':', '=') until first digit.
+    /// Returns false if no version-like token is found.
+    /// </summary>
+    private static bool TryExtractVersion(ReadOnlySpan<char> haystack, out Range range)
+    {
+        range = default;
+
+        // Limit search window to avoid scanning entire UA string unnecessarily
+        const int Window = 128;
+        if (haystack.Length >= Window)
+        {
+            haystack = haystack.Slice(0, Window);
+        }
+
+        int i = 0;
+        for (; i < haystack.Length; ++i)
+        {
+            char c = haystack[i];
+            if (char.IsBetween(c, '0', '9'))
+            {
+                break;
+            }
+        }
+
+        int s = i;
+        haystack = haystack.Slice(i + 1);
```

Comment:
We have a test for an invalid useragent. With this code it throws an ArgumentOutOfRangeException here

Reply:
Wanted to add a test and create a pr by myself. But couldn't get the solution to build. (Have not installed .net10 preview on my machine)

Reply:
What`s "Mozilla/ (iPhone; U; CPU iPhone OS  like Mac OS X) AppleWebKit/ (KHTML, like Gecko) Version/ Mobile/ Safari/" for a user agent?
I cant find anything matching online.

Reply:
Yeah that is a invalid one. 

We have a test which tries to parse this one (can not remember how we came up with this one...) 

We use it to test that we do not proceed when the ua is invalid. 

I am ok to change my test, but the parser did not throw before and now throws an exception 🤷